gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -704,36 +708,114 @@ public boolean bulkApiLogin() throws Exception {
     return success;
   }
 
+  public JobIdAndBatchIdResultIdList getQueryResultIdsPkChunkingFetchOnly(String jobId, String batchIdListStr) {
```

Comment:
Please add javadoc for this public method. If this is only for testing then mention that in the javadoc.

Reply:
added javadoc
----
   * same as getQueryResultIdsPkChunking but the arguments are different.
   * this function can take existing batch ids to return JobIdAndBatchIdResultIdList
   * It is for test/debug. developers may want to skip execute query on SFDC, use a list of existing batch ids